include/boost/circular_buffer/base.hpp

```diff
@@ -1449,6 +1449,42 @@ class circular_buffer
         BOOST_CATCH_END
     }
 
+    template <class ...Args>
+    void emplace_back_impl(BOOST_FWD_REF(Args) ...args) {
+        if (full()) {
+            if (empty())
+                return;
+            replace(m_last, value_type(::boost::forward<Args>(args)...));
+            increment(m_last);
+            m_first = m_last;
+        } else {
+            boost::container::allocator_traits<Alloc>::construct(m_alloc, cb_details::to_address(m_last), ::boost::forward<Args>(args)...);
+            increment(m_last);
+            ++m_size;
+        }
+    }
+
+    template <class ...Args>
+    void emplace_front_impl(BOOST_FWD_REF(Args) ...args) {
+        BOOST_TRY {
+            if (full()) {
+                if (empty())
+                    return;
+                decrement(m_first);
+                replace(m_first, value_type(::boost::forward<Args>(args)...));
+                m_last = m_first;
+            } else {
+                decrement(m_first);
+                boost::container::allocator_traits<Alloc>::construct(m_alloc, cb_details::to_address(m_first), ::boost::forward<Args>(args)...);
```

Comment:
The library no longer has a dependency on Boost.Container for `boost::container::allocator_traits`. Instead use `cb_details::allocator_traits` which is defined in <boost/circular_buffer/allocators.hpp>.